windows/RNFS/RNFSManager.h

```diff
@@ -0,0 +1,180 @@
+
+    REACT_METHOD(readFile); // Implemented
+    winrt::fire_and_forget readFile(std::string filePath, RN::ReactPromise<std::string> promise) noexcept;
+
+    REACT_METHOD(read); // Implemented
+    winrt::fire_and_forget read(
+        std::string filePath,
+        uint32_t length,
+        uint64_t position,
+        RN::ReactPromise<std::string> promise) noexcept;
+
+    REACT_METHOD(hash); // Implemented
+    winrt::fire_and_forget hash(std::string filepath, std::string algorithm, RN::ReactPromise<std::string> promise) noexcept;
+
+    REACT_METHOD(writeFile); // Implemented
+    winrt::fire_and_forget writeFile(
+        std::string filePath,
+        std::string base64Content,
+        RN::JSValueObject options,
+        RN::ReactPromise<void> promise) noexcept;
+
+    REACT_METHOD(appendFile); // Implemented, no unit tests
+    winrt::fire_and_forget appendFile(
+        std::string filepath,
+        std::string base64Content,
+        RN::ReactPromise<void> promise
+    ) noexcept;
+
+
+    REACT_METHOD(write); // Implemented
+    winrt::fire_and_forget write(
+        std::string filePath,
+        std::string base64Content,
+        int position,
+        RN::ReactPromise<void> promise) noexcept;
+
+
+    REACT_METHOD(downloadFile); // DOWNLOADER
+    winrt::fire_and_forget downloadFile(RN::JSValueObject options, RN::ReactPromise<RN::JSValueObject> promise) noexcept;
+
+    REACT_METHOD(uploadFiles); // DOWNLOADER
+    winrt::fire_and_forget uploadFiles(RN::JSValueObject options, RN::ReactPromise<RN::JSValueObject> promise) noexcept;
+
+    REACT_METHOD(touch); // Implemented
+    void touch(std::string filepath, int64_t mtime, int64_t ctime, bool modifyCreationTime, RN::ReactPromise<std::string> promise) noexcept;
+
+    REACT_EVENT(TimedEvent, L"TimedEventCpp");
+    std::function<void(int)> TimedEvent;
+
+private:
+    void splitPath(const std::string& fullPath, winrt::hstring& directoryPath, winrt::hstring& fileName) noexcept;
+
+    winrt::Windows::Foundation::IAsyncAction ProcessDownloadRequestAsync(RN::ReactPromise<RN::JSValueObject> promise,
+        winrt::Windows::Web::Http::HttpRequestMessage request, std::wstring_view filePath, int32_t jobId, int64_t progressInterval, int64_t progressDivider);
+
+    winrt::Windows::Foundation::IAsyncAction ProcessUploadRequestAsync(RN::ReactPromise<RN::JSValueObject> promise, RN::JSValueObject& options,
+        winrt::Windows::Web::Http::HttpMethod httpMethod, RN::JSValueArray const& files, int32_t jobId, uint64_t totalUploadSize);
+
+    winrt::fire_and_forget copyFolderHelper(
+        winrt::Windows::Storage::StorageFolder src,
+        winrt::Windows::Storage::StorageFolder dest) noexcept;
+
+    constexpr static int64_t UNIX_EPOCH_IN_WINRT_INTERVAL = 11644473600 * 10000000;
+     
+    const std::map<std::string, std::function<CryptographyCore::HashAlgorithmProvider()>> availableHashes{
```

Comment:
this can be constexpr / static const? also no need for a map where an unordered_map or even a `vector<pair<string, function>>` will do :) it's only 5 elements